other_packages/python/tiledbsoma_ml/src/tiledbsoma_ml/pytorch.py

```diff
@@ -0,0 +1,1173 @@
+    produces equal sized ``X`` and ``obs`` data, in the form of a :class:``numpy.ndarray`` and
+    :class:`pandas.DataFrame`.
+
+    Private base class for subclasses of :class:`torch.utils.data.IterableDataset` and
+    :class:`torchdata.datapipes.iter.IterDataPipe`. Refer to :class:`ExperimentAxisQueryIterableDataset`
+    and `ExperimentAxisQueryDataPipe` for more details on usage.
+
+    Lifecycle:
+        experimental
+    """
+
+    def __init__(
+        self,
+        query: soma.ExperimentAxisQuery,
+        X_name: str,
+        obs_column_names: Sequence[str] = ("soma_joinid",),
+        batch_size: int = 1,
+        shuffle: bool = True,
+        io_batch_size: int = 2**16,
+        shuffle_chunk_size: int = 64,
+        seed: int | None = None,
+        use_eager_fetch: bool = True,
+    ):
+        """
+        Construct a new ``ExperimentAxisQueryIterable``, suitable for use with :class:`torch.utils.data.DataLoader`.
+
+        The resulting iterator will produce a 2-tuple containing associated slices of ``X`` and ``obs`` data, as
+        a NumPy ``ndarray`` and a Pandas ``DataFrame`` respectively.
+
+        Args:
+            query:
+                A :class:`tiledbsoma.ExperimentAxisQuery`, defining the data which will be iterated over.
+            X_name:
+                The name of the X layer to read.
+            obs_column_names:
+                The names of the ``obs`` columns to return. At least one column name must be specified.
+                Default is ``('soma_joinid',)``.
+            batch_size:
+                The number of rows of ``X`` and ``obs`` data to return in each iteration. Defaults to ``1``. A value of
+                ``1`` will result in :class:`torch.Tensor` of rank 1 being returns (a single row); larger values will
+                result in :class:`torch.Tensor`\ s of rank 2 (multiple rows).
+
+                Note that a ``batch_size`` of 1 allows this ``IterableDataset`` to be used with :class:`torch.utils.data.DataLoader`
+                batching, but you will achieve higher performance by performing batching in this class, and setting the ``DataLoader``
+                batch_size parameter to ``None``.
+            shuffle:
+                Whether to shuffle the ``obs`` and ``X`` data being returned. Defaults to ``True``.
+            io_batch_size:
+                The number of ``obs``/``X`` rows to retrieve when reading data from SOMA. This impacts two aspects of
+                this class's behavior: 1) The maximum memory utilization, with larger values providing
+                better read performance, but also requiring more memory; 2) The number of rows read prior to shuffling
+                (see ``shuffle`` parameter for details). The default value of 131,072 provides high performance, but
+                may need to be reduced in memory limited hosts (or where a large number of :class:`DataLoader` workers
+                are employed).
```

Comment:
```suggestion
                The number of ``obs``/``X`` rows to retrieve when reading data from SOMA. This impacts: 
                1. Maximum memory utilization, larger values provide better read performance, but require more memory. 
                2. The number of rows read prior to shuffling (see the ``shuffle`` parameter for details). 
                The default value of 131,072 provides high performance but may need to be reduced in memory-limited hosts
                or when using a large number of :class:`DataLoader` workers.
```

I'm not super-familiar with sphinx's markup but it [looks like](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#lists-and-quote-like-blocks) lists are supported—thought it might help make this section a little easier to digest.